torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -304,8 +325,10 @@ static void cacheAllocatorRegisterHook(
   std::lock_guard<std::mutex> lock(ncclCommDevIdxMapMutex);
   for (auto& [ncclComm, _] : ncclCommDevIdxMap) {
     if (te.device_ == ncclComm->getDeviceIndex()) {
-      // NOLINTNEXTLINE(performance-no-int-to-ptr)
-      ncclComm->registerSegment(reinterpret_cast<void*>(te.addr_), te.size_);
+      if (shouldAllCommunicatorsRegisterAllTensors()) {
```

Comment:
This check is currently useless (since these hooks are only ever attached and invoked if that flag is true), but in a later PT I plan to use these hooks for other purposes as well, hence this is future-proofing.